pkg/policy/mapstate.go

```diff
@@ -687,15 +687,17 @@ func (e *MapStateEntry) HasDependent(key Key) bool {
 	return ok
 }
 
-// HasSameOwners returns true if both MapStateEntries
-// have the same owners as one another (which means that
-// one of the entries is redundant).
-func (e *MapStateEntry) HasSameOwners(bEntry *MapStateEntry) bool {
-	if len(e.owners) != len(bEntry.owners) {
+// CoversOwners returns true if the receiver MapStateEntry
+// has all the owners that the argument MapStateEntry has.
+func (e *MapStateEntry) CoversOwners(bEntry *MapStateEntry) bool {
+	if bEntry == nil || len(bEntry.owners) == 0 {
+		return true
+	}
+	if len(e.owners) < len(bEntry.owners) {
 		return false
 	}
-	for owner := range e.owners {
-		if _, ok := bEntry.owners[owner]; !ok {
+	for _, owner := range bEntry.owners {
```

Comment:
This is reintroducing a previously fixed bug. 'owners' is a map:
```
owners map[MapStateOwner]struct{}
```
So now the value `struct{}{}` is used as the 'owner', which is never found from `e.owners`, so this function will always return `false`.